flink-streaming-java/src/main/java/org/apache/flink/streaming/api/environment/CheckpointConfig.java

```diff
@@ -1016,6 +1016,10 @@ public void configure(ReadableConfig configuration) {
         configuration
                 .getOptional(CheckpointingOptions.CHECKPOINTS_DIRECTORY)
                 .ifPresent(this::setCheckpointStorage);
+        // reset checkpoint storage for backward compatibility
+        configuration
+                .getOptional(CheckpointingOptions.CHECKPOINT_STORAGE)
+                .ifPresent(ignored -> this.storage = null);
```

Comment:
It's a bit weird here to set and reset the checkpointStorage back and forth. Maybe do not invoke `setCheckpointStorage` but direct do `configuration.set(CheckpointingOptions.CHECKPOINTS_DIRECTORY,...)`?